File: examples/ScreenPlateWell/listplates.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
import omero
from omero_sys_ParametersI import ParametersI  # Temporary

c = omero.client()
s = c.createSession()
q = s.getQueryService()

QUERY = ("select p from Plate p left outer join fetch p.wells w"
         " left outer join fetch w.wellSamples s"
         " left outer join fetch s.image where p.id = :id")

filter = omero.sys.Filter()
plates = q.findAll('Plate', filter)


print 'Plate:'
for i in range(len(plates)):
    params = ParametersI()
    params.addId(plates[i].getId().getValue())
    plate = q.findByQuery(QUERY, params)
    print '    %d %s ' % (i, plate.getName().getValue())
    wells = {}

    for well in plate.copyWells():
        row = well.getRow() and well.getRow().getValue() or -1
        col = well.getColumn() and well.getColumn().getValue() or -1
        if row in wells:
            row_list = wells[row]
        else:
            row_list = []
            wells[row] = row_list
        row_list.append(col)
        row_list.sort()

    rows = list(wells.keys())
    rows.sort()

    for row in rows:
        msg = '        '
        col_list = list(wells[row])
        col_list.sort()
        for col in col_list:
            msg += '%2dx%2d ' % (row, col)
        print msg

File: examples/ScreenPlateWell/listplates.m
import omero.*

c = omero.client();
s = c.createSession();
q = s.getQueryService();

QUERY = 'select p from Plate p left outer join fetch p.wells w left outer join fetch w.wellSamples s left outer join fetch s.image where p.id = :id';

filter = omero.sys.Filter();
plates = q.findAll('Plate', filter);


disp('Plate:');
for i=0:plates.size()-1
    params = omero.sys.ParametersI();
    params.addId(plates.get(i).getId().getValue());
    plate = q.findByQuery(QUERY, params);
    disp(sprintf('    %d %s %s', i, char(plate.getName().getValue())))
    wells = java.util.HashMap();

    well_it = plate.copyWells().listIterator();
    while well_it.hasNext()
        well = well_it.next();
        row = well.getRow().getValue();
        col = well.getColumn().getValue();
        if wells.containsKey(row)
            row_list = wells.get(row);
        else
            row_list = java.util.ArrayList();
            wells.put(row, row_list);
        end
        row_list.add(col);
        java.util.Collections.sort(row_list);
    end

    rows = java.util.ArrayList(wells.keySet());
    java.util.Collections.sort(rows);

    row_it = rows.listIterator();
    while row_it.hasNext();
        row = row_it.next();
        msg = '        ';
        col_list = java.util.ArrayList(wells.get(row));
        java.util.Collections.sort(col_list);
        col_it = col_list.listIterator();
        while col_it.hasNext()
            col = col_it.next();
            msg = [msg, sprintf('%2dx%2d ', row, col)];
        end
        disp(msg)
    end
end
